pdfviewer-library/src/main/java/xyz/guutong/androidpdfviewer/PdfViewActivity.java

```diff
@@ -127,31 +131,53 @@ public boolean onOptionsItemSelected(MenuItem item) {
         if (i == MENU_CLOSE) {
             finish();
         } else if (i == MENU_SHARE) {
-            AlertDialog.Builder builder = new AlertDialog.Builder(this);
-            CharSequence[] itemsAlert = {"Copy link", "Open browser"};
-
-            builder.setItems(itemsAlert, new DialogInterface.OnClickListener() {
-                @Override
-                public void onClick(DialogInterface dialog, int itemIndex) {
-                    final int COPY_LINK = 0;
-                    final String label = "URL";
-
-                    if (itemIndex == COPY_LINK) {
-                        ClipboardManager clipboard = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
-                        ClipData clip = ClipData.newPlainText(label, pdfUrl);
-                        clipboard.setPrimaryClip(clip);
-                        return;
-                    }
 
-                    Intent intentBrowser = new Intent(Intent.ACTION_VIEW);
-                    intentBrowser.setData(Uri.parse(pdfUrl));
-                    startActivity(intentBrowser);
+            if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, 200);
+            } else {
+                saveFile();
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
+        switch (requestCode) {
+            case 200: {
+                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    saveFile();
                 }
-            });
-            builder.show();
+                return;
+            }
+        }
+    }
+
+    private void saveFile(){
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+
+        try {
+            int bytesum = 0;
+            int byteread = 0;
+            File oldfile = new File(getCacheDir(), FileUtil.extractFileNameFromURL(pdfUrl));
+
+            InputStream inStream = new FileInputStream(oldfile.getAbsoluteFile());
+            FileOutputStream fs = new FileOutputStream(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + "/" + FileUtil.extractFileNameFromURL(pdfUrl));
+            byte[] buffer = new byte[1444];
+            while ((byteread = inStream.read(buffer)) != -1) {
+                bytesum += byteread;
+                fs.write(buffer, 0, byteread);
+            }
+            inStream.close();
+            fs.close();
+
+            builder.setMessage("File was saved in Downloads");
```

Comment:
A good practice is to save the strings in the resource. We should have a context for that.